### ISSUE TITLE
Question - Github as a backend - Storage limit

### ISSUE THREAD

**Issue by cyboolo** [simplaio/simpla#93]
Hi,

It's not an issue... not yet

I was wondering... simpla uses github as a backend to store data... it's ok

BUT, there is a storage limit of 1 Gb on github repository... as I am storing images and texts from multiple domains from 1 repository... how do you think I should scale

I saw git lfs but I don't understand how it works and I am not sure that pointers created will work with simpla...

So it's just an interrogation right now... but could become a big issue or a big limitation of using simpla in my case if I don't know how to handle this in the future.

**Comment by madeleineostoja** [comment]
The first step to solving this would be addressing #77, so that people could use a different storage backend if they hit limits with Github. 

I'm not sure how LFS works either, but that could be an option. As part of #77 we could also split out content and assets (ie: images, etc), which we've been wanting to do on the SDK anyway.

**Comment by cyboolo** [comment]
Github support:
Hi Olivier,

Thanks for reaching out! GitHub's main use is for version control, and it is not intended for use as a backend or content distribution network for large amounts of files. If you need to store and distribute a lot of files, we recommend using a dedicated CDN for that purpose instead.

Data packs only apply to Git LFS – we don't offer any storage upgrades outside of LFS.

I hope that helps clarify things! Feel free to let us know if you have any other questions.

All the best,
Ari

**Comment by cyboolo** [comment]
So, in my opinion, it would be priority number 1 to have github alternative. If not that means that apps made with simpla can not be scaled so that means plans limitations, storage survey (not possible with github api),...

And i tried git lfs but don’t work outside of terminal push... or i didn’t understand how to upload or push to use lfs... and anyway i am quite sure that pointers created will create issues with simpla.

For instance, my app is built with sashido.io and parse server, similar to firebase. I could really easily store json files and uploaded images on my database and create functions to set, update, delete, query... but i don’t know how to link to your code.

But github limitation is really big issue for me. It means I have 1 Gb for all users! I am coding to create one github repo asociated to one netlify site for each of my users to have 1 Gb/user but 1 Gb is nothing.

Hope you find a good alternative fastly or at least some kind of adapter that allows simpla user to put _content data where they want.

If not, that means simpla could be use for very simple static sites with few pages.

**Comment by madeleineostoja** [comment]
Yep, as per #77, custom storage adapters are on the roadmap. FWIW, 1gb of content is a fair bit, NetlifyCMS (which powers sites like Smashing Magazine) runs on Github as well. But still a valid concern, which will be addressed by opening up storage as an adapter rather than hard-coding Github.

**Comment by cyboolo** [comment]
I managed via netlify and github api to create one specific site and repo for each of my user.

I import an existing repo of my model static site on each new repo. This model has bower libraries included... quite heavy... Could I install these by npm build on deploying on netlify or via bower install...to avoid to have to import those libraries each time?

**Comment by madeleineostoja** [comment]
Yep, Netlify will autoamtically install Bower dependencies for you on build
